builtin/providers/aws/resource_aws_instance.go

```diff
@@ -469,6 +469,13 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("public_ip", instance.PublicIpAddress)
 	d.Set("private_dns", instance.PrivateDnsName)
 	d.Set("private_ip", instance.PrivateIpAddress)
+
+	if instance.IamInstanceProfile != nil {
+		d.Set("iam_instance_profile", iamInstanceProfileArnToName(instance.IamInstanceProfile.Arn))
+	} else if _, ok := d.GetOk("iam_instance_profile"); ok {
```

Comment:
As far as I know, setting it to the empty string only when ``GetOk`` returns true is functionally equivalent to *always* setting it to the empty string, since for string schemata that is the "zero value" anyway. So I think this could just be a simple ``else`` block, rather than an ``else if``.